src/metrics.rs

```diff
@@ -89,68 +93,114 @@ pub(crate) fn processing_time(direction: Direction) -> Histogram {
     PROCESSING_TIME.with_label_values(&[direction.label()])
 }
 
-pub(crate) fn bytes_total(direction: Direction) -> IntCounter {
+pub(crate) fn bytes_total(direction: Direction, asn: Option<&IpNetEntry>) -> IntCounter {
     static BYTES_TOTAL: Lazy<IntCounterVec> = Lazy::new(|| {
         prometheus::register_int_counter_vec_with_registry! {
             prometheus::opts! {
                 "bytes_total",
                 "total number of bytes",
             },
-            &[Direction::LABEL],
+            &[Direction::LABEL, ASN_LABEL, PREFIX_LABEL],
             registry(),
         }
         .unwrap()
     });
 
-    BYTES_TOTAL.with_label_values(&[direction.label()])
+    BYTES_TOTAL.with_label_values(&[
+        direction.label(),
+        &asn.map(|asn| asn.r#as.to_string()).unwrap_or_default(),
+        asn.map(|asn| &*asn.prefix).unwrap_or_default(),
+    ])
 }
 
-pub(crate) fn errors_total(direction: Direction, display: &str) -> IntCounter {
+pub(crate) fn errors_total(
+    direction: Direction,
+    display: &str,
+    asn: Option<&IpNetEntry>,
+) -> IntCounter {
     static ERRORS_TOTAL: Lazy<IntCounterVec> = Lazy::new(|| {
         prometheus::register_int_counter_vec_with_registry! {
             prometheus::opts! {
                 "errors_total",
                 "total number of errors sending packets",
             },
-            &[Direction::LABEL, "display"],
+            &[Direction::LABEL, "display", ASN_LABEL, PREFIX_LABEL],
+            registry(),
+        }
+        .unwrap()
+    });
+
+    ERRORS_TOTAL.with_label_values(&[
+        direction.label(),
+        display,
+        &asn.map(|asn| asn.r#as.to_string()).unwrap_or_default(),
+        asn.map(|asn| &*asn.prefix).unwrap_or_default(),
+    ])
+}
+
+pub(crate) fn packet_jitter(direction: Direction, asn: Option<&IpNetEntry>) -> IntGauge {
+    static PACKET_JITTER: Lazy<IntGaugeVec> = Lazy::new(|| {
+        prometheus::register_int_gauge_vec_with_registry! {
+            prometheus::opts! {
+                "packet_jitter",
+                "The time between new packets",
+            },
+            &[Direction::LABEL, ASN_LABEL, PREFIX_LABEL],
             registry(),
         }
         .unwrap()
     });
 
-    ERRORS_TOTAL.with_label_values(&[direction.label(), display])
+    PACKET_JITTER.with_label_values(&[
+        direction.label(),
+        &asn.map(|asn| asn.r#as.to_string()).unwrap_or_default(),
+        asn.map(|asn| &*asn.prefix).unwrap_or_default(),
+    ])
 }
 
-pub(crate) fn packets_total(direction: Direction) -> IntCounter {
+pub(crate) fn packets_total(direction: Direction, asn: Option<&IpNetEntry>) -> IntCounter {
```

Comment:
(adding review here ,since this is the one that will get merged)

Looks good - only thing to add would be some docs to metrics.md to document the new metrics, otherwise, good to merge 👍🏻